### PR TITLE
extra-container: init at 0.8

### PR DESCRIPTION
**Pull request by erikarvstedt**
[extra-container](https://github.com/erikarvstedt/extra-container/) is a tool for running declarative NixOS containers without host system rebuilds.

cc @Lassulus

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/virtualization/extra-container/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchFromGitHub {
+    owner = "erikarvstedt";
+    repo = pname;
+    rev = version;
+    sha256 = "12xqa11v583ajdv51g1833rxvrndmly9h4r62wc3llm8xs6k7ais";
+  };
+
+  buildCommand = ''
+    install -D $src/extra-container $out/bin/extra-container
+    patchShebangs $out/bin
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/extra-container/default.nix:16]:
```suggestion
    patchShebangs $out/bin
    install -D $src/extra-container $out/bin/extra-container
```

File: pkgs/tools/virtualization/extra-container/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      -e "s|LOCALE_ARCHIVE=.*|LOCALE_ARCHIVE=${glibcLocales}/lib/locale/locale-archive|" \
+      -e "2i$scriptPath" \
+      $out/bin/extra-container
+  '';
+
+  meta = with lib; {
+    description = "Run declarative containers without full system rebuilds";
+    homepage = https://github.com/erikarvstedt/extra-container;
+    license = licenses.mit;
+    maintainers = [ maintainers.earvstedt ];
```
Comment (SuperSandro2000) [inline pkgs/tools/virtualization/extra-container/default.nix:34]: You are missing platforms.

File: pkgs/tools/virtualization/extra-container/default.nix (hunk 1 above)
Comment (erikarvstedt) [reply pkgs/tools/virtualization/extra-container/default.nix:16]: We have to copy the script file to `bin` before patching it.

File: pkgs/tools/virtualization/extra-container/default.nix (hunk 2 above)
Comment (erikarvstedt) [reply pkgs/tools/virtualization/extra-container/default.nix:35]: Fixed.

File: pkgs/tools/virtualization/extra-container/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/virtualization/extra-container/default.nix:16]: Can you explain why?

Comment (erikarvstedt) [reply pkgs/tools/virtualization/extra-container/default.nix:16]: `patchShebangs $out/bin` (line 1) would fail because `$out/bin` doesn't exist.

Comment (SuperSandro2000) [reply pkgs/tools/virtualization/extra-container/default.nix:16]: I mena ``patchShebangs $src/extra-container``.

Comment (erikarvstedt) [reply pkgs/tools/virtualization/extra-container/default.nix:16]: In this case it will fail because `$src` is a read-only store path.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102354` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>extra-container</li>
  </ul>
</details>

**Comment by veehaitch** [comment]
@erikarvstedt Thanks for this PR. Could you please resolve the conflict and update the PR to use the latest version?

**Comment by Lassulus** [comment]
I have taken the liberty to fix the merge conflict and update to the newest version

**Comment by erikarvstedt** [comment]
I've released a new version including the latest changes that add compatibility with nixpkgs unstable.

**Comment by erikarvstedt** [comment]
Test the NixOS module with:
```
cfg="{
  containers.tmp.config = {
    programs.extra-container.enable = true;
    nix.nixPath = [ \"nixpkgs=$(realpath $(nix-instantiate --eval -E '<nixpkgs>'))\" ];
  };
}"
cfg2='{ containers.tmp.config = {}; }'
extra-container shell --nixpkgs-path /path/to/nixpkgs -E "$cfg" --run c extra-container shell -E "$cfg2" --run echo success
```
This builds a container with `programs.extra-container` enabled and runs `extra-container` within this container.

**Review by Lassulus — APPROVED** [review]

**Pull request merged**